src/jshint.js

```diff
@@ -4247,12 +4305,12 @@ var JSHINT = (function() {
     x.lbp = 25;
   }(prefix("yield", function() {
     var prev = state.tokens.prev;
-    if (state.inESNext(true) && !state.funct["(generator)"]) {
+    if (state.inES6(true) && !state.funct["(generator)"]) {
```

Comment:
generator functions aren't in 6.. but I guess the plan for this is for v2.x.. meaning it cannot break?
Still might be good to differentiate *now* e.g. esnext means esversion = 7 and this requires es7?